ci/libvcx.dockerfile

```diff
@@ -1,81 +1,76 @@
-
-# Move the binary to system library
-USER root
-RUN mv /home/indy/libvcx/target/release/*.so /usr/lib
+ENV RUST_LOG=warning
 
+RUN addgroup -g $GID indy && adduser -u $UID -D -G indy indy
 
-# Create a new build stage and copy outputs from BASE
-FROM ubuntu:16.04
-
-RUN apt-get update && \
-    apt-get install -y \
-      libssl-dev \
-      apt-transport-https \
-      ca-certificates \
-      curl \
-      build-essential
-
-RUN useradd -ms /bin/bash -u 1000 indy
+RUN apk update && apk upgrade && \
+    apk add --no-cache \
+        build-base \
+        cargo \
+        git \
+        libsodium-dev \
+        libzmq \
+        openssl-dev \
+        rust \
+        sqlite-dev \
+        zeromq-dev
 
+USER indy
 WORKDIR /home/indy
 
-COPY --from=BASE /var/lib/dpkg/info /var/lib/dpkg/info
-COPY --from=BASE /usr/lib/x86_64-linux-gnu /usr/lib/x86_64-linux-gnu
-COPY --from=BASE /usr/local /usr/local
-
-COPY --from=BASE --chown=indy /usr/lib/libindy.so /usr/lib/libindy.so
-COPY --from=BASE --chown=indy /usr/lib/libvcx.so /usr/lib/libvcx.so
-COPY --from=BASE --chown=indy /usr/lib/libnullpay.so /usr/lib/libnullpay.so
-COPY --from=BASE --chown=indy /usr/lib/libindystrgpostgres.so /usr/lib/libindystrgpostgres.so
+RUN git clone $INDYSDK_REPO && \
+    cd indy-sdk && git checkout $INDYSDK_REVISION
 
-COPY --from=BASE --chown=indy /home/indy/libvcx ./libvcx
-COPY --from=BASE --chown=indy /home/indy/wrappers/node ./wrappers/node
+RUN cargo build --release --manifest-path=$INDYSDK_PATH/libindy/Cargo.toml
 
-# Install node
-ARG NODE_VER=8.x
-RUN curl -sL https://deb.nodesource.com/setup_${NODE_VER} | bash -
-RUN apt-get install -y nodejs
-
-RUN chown -R indy . 
+USER root
+RUN mv $INDYSDK_PATH/libindy/target/release/libindy.so /usr/lib
 
 USER indy
+RUN cargo build --release --manifest-path=$INDYSDK_PATH/vcx/libvcx/Cargo.toml
+RUN cargo build --release --manifest-path=$INDYSDK_PATH/libnullpay/Cargo.toml
+RUN cargo build --release --manifest-path=$INDYSDK_PATH/experimental/plugins/postgres_storage/Cargo.toml
 
-# TODO: Just copy the binary and add to path
-ARG RUST_VER=1.40.0
-RUN curl https://sh.rustup.rs -sSf | sh -s -- -y --default-toolchain ${RUST_VER}
-ENV PATH /home/indy/.cargo/bin:$PATH
-
-LABEL org.label-schema.schema-version="1.0"
+USER root
+RUN mv $INDYSDK_PATH/vcx/libvcx/target/release/libvcx.so .
+RUN mv $INDYSDK_PATH/libnullpay/target/release/libnullpay.so .
+RUN mv $INDYSDK_PATH/experimental/plugins/postgres_storage/target/release/libindystrgpostgres.so .
+
+FROM alpine:3.12
+
+ARG UID=1000
+ARG GID=1000
+
+RUN addgroup -g $GID node && adduser -u $UID -D -G node node
+
+COPY --from=builder /usr/lib/libindy.so /home/indy/lib*.so /usr/lib/
+
+RUN echo '@alpine38 http://dl-cdn.alpinelinux.org/alpine/v3.8/main' >> /etc/apk/repositories
+
+RUN apk update && apk upgrade
+RUN apk add --no-cache \
+        bash \
+        g++ \
+        gcc \
+        git \
+        libsodium-dev \
+        libzmq \
+        make \
+        nodejs@alpine38 \
```

Comment:
I think it would be useful to make the node version configurable via an ARG. If not possible, let's remove it from the `main.yml` as well.